crates/ruma-client-api/src/error/kind_serde.rs

```diff
@@ -214,6 +227,20 @@ impl<'de> Visitor<'de> for ErrorKindVisitor {
             ErrCode::CannotOverwriteMedia => ErrorKind::CannotOverwriteMedia,
             #[cfg(feature = "unstable-msc3575")]
             ErrCode::UnknownPos => ErrorKind::UnknownPos,
+            ErrCode::UrlNotSet => ErrorKind::UrlNotSet,
+            ErrCode::BadStatus => ErrorKind::BadStatus {
+                status: status
+                    .map(|s| {
+                        from_json_value::<u16>(s)
+                            .map_err(de::Error::custom)?
+                            .try_into()
+                            .map_err(de::Error::custom)
+                    })
+                    .transpose()?,
+                body: body.map(|b| from_json_value(b).map_err(de::Error::custom)).transpose()?,
```

Comment:
```suggestion
                body: body.map(from_json_value).transpose().map_err(de::Error::custom)?,
```

Reply:
A bit shorter and has the same meaning, right? I don't mind leaving the existing form in if you prefer though.

Reply:
I prefer your version. My brain must have been in automatic mode with the lines above.